xma_data_processor.py

```diff
@@ -0,0 +1,374 @@
+"""Converts data from XMALab into the format useful for DLC training."""
+
+import os
+
+import blend_modes
+import cv2
+import numpy as np
+import pandas as pd
+from ruamel.yaml import YAML
+
+
+class XMADataProcessor:
+    """Converts data from XMALab into the format useful for DLC training."""
+
+    def __init__(self, config):
+        self._config = config
+        self._mode = config["tracking_mode"]
+
+    def get_bodyparts_from_xma(
+        self, path_to_trial, split_markers=False, crossed_markers=False
+    ):
+        """Pulls the names of the XMAlab markers from the 2Dpoints file"""
+
+        csv_path = [file for file in os.listdir(path_to_trial) if file[-4:] == ".csv"]
+        if len(csv_path) > 1:
+            raise FileExistsError(
+                "Found more than 1 CSV file for trial: " + path_to_trial
+            )
+        if len(csv_path) <= 0:
+            raise FileNotFoundError(
+                "Couldn't find a CSV file for trial: " + path_to_trial
+            )
+        trial_csv = pd.read_csv(
+            path_to_trial + "/" + csv_path[0],
```

Comment:
May want to consider using os.path.join here... typically this works fine on all systems, just don't want to run into any strange edge cases.

Reply:
done (uploading new commits momentarily)